hail/python/hailtop/batch/batch.py

```diff
@@ -396,11 +397,12 @@ def _new_job_resource_file(self, source, value=None):
         self._resource_map[jrf._uid] = jrf  # pylint: disable=no-member
         return jrf
 
-    def _new_input_resource_file(self, input_path, value=None):
+    def _new_input_resource_file(self, input_path, root=None):
         self._backend.validate_file_scheme(input_path)
-        if value is None:
-            value = f'{secret_alnum_string(5)}/{os.path.basename(input_path.rstrip("/"))}'
-        irf = _resource.InputResourceFile(value)
+        root = root or secret_alnum_string(5)
+        # Take care not to include a SAS token query string in the local name.
+        file_name, _ = AzureAsyncFS.get_name_parts(input_path.rstrip("/"))
```

Comment:
For clarity, could we do something like (not married to the names)

```python
if RouterAsyncFS.blob_cloud(input_path) == 'azure':
    cleaned_path = AzureAsyncFS.strip_sas_token(input_path)
```

Reply:
added the cloud type check - would prefer to keep the `get_name_parts` signature as-is since there are other callers that need both parts, but let me know if you think it's better to have an additional wrapper method on `AzureAsyncFS`